core/base/model/TomlFile.py

```diff
@@ -0,0 +1,316 @@
+			section.addConfig(key=key, value=val, commented=False)
+
+
+	def addEmptinessIfNeeded(self):
+		"""
+		Add a space between the last existing section and a new one
+		"""
+		if self._loaded:
+			try:
+				self._data[list(self._data.keys())[-1]].addEmptiness()
+			except Exception:
+				# No need to add new emptiness
+				pass
+
+
+	def __delitem__(self, key: str):
+		if key in self._data:
+			del self._data[key]
+
+
+	def __contains__(self, item) -> bool:
+		return item in self._data
+
+
+	def values(self) -> ValuesView:
+		return self._data.values()
+
+
+	def items(self) -> ItemsView:
+		return self._data.items()
+
+
+	def get(self, key, default: Any) -> Any:
+		return self._data.get(key, default)
+
+
+class Comment:
+
+	def __init__(self, comment: str):
+		self.comment = comment.strip()
+
+
+	def __repr__(self):
+		return self.comment
+
+
+	def __str__(self):
+		return self.comment
+
+
+class Section(dict):
+
+	def __init__(self, name: str):
+		super().__init__()
+		self.name = name
+		self.data: Dict[str, Union[Comment, Config, Emptiness]] = dict()
+		self._comments = 0
+		self._whites = 0
+
+
+	def __len__(self) -> int:
+		return len(self.data)
+
+
+	def __iter__(self):
+		return iter(self.data)
+
+
+	def __setitem__(self, key: str, value: Any):
+		if key in self.data:
+			self.data[key].value = value
+			self.data[key].uncomment()
+		else:
+			self.data[key] = Config(key, value)
+
+
+	def __getitem__(self, key: str) -> Any:
+		if key in self.data:
+			return self.data[key].value
+
+		conf = Config(key, '')
+		self.data[conf.name] = conf
+		return conf.value
+
+
+	def __delitem__(self, key: str):
+		del self.data[key]
+
+
+	def __contains__(self, item) -> bool:
+		return item in self.data
+
+
+	def __repr__(self) -> dict:
+		return self.data
+
+
+	def __str__(self):
+		response = f'* Section "{self.name}" with {len(self.data) - self._comments} configurations:\n'
+		for conf in self.data.values():
+			if isinstance(conf, Comment) or isinstance(conf, Emptiness):
+				continue
+
+			response += f' - {conf.name} = {conf.value} | Commented: {conf.commented}\n'
+		return dedent(response)
+
+
+	def values(self) -> ValuesView:
+		return self.data.values()
+
+
+	def items(self) -> ItemsView:
+		return self.data.items()
+
+
+	# noinspection PyMethodOverriding
+	def get(self, key, default: Any) -> Any:
+		return self.data.get(key, default)
+
+
+	def addComment(self, comment: Comment):
+		self._comments += 1
+		self.data[f'comment_{self._comments}'] = comment
+
+
+	def addConfig(self, key: str, value: Any, commented: bool):
+		config = Config(key, value, commented)
+		self.data[config.name] = config
+
+
+	def addEmptiness(self):
+		self._whites += 1
+		self.data[f'whites_{self._whites}'] = Emptiness()
+
+
+class Config:
+
+	def __init__(self, key: str, value: Any, commented: bool = False):
+		self.name = key
+		self.commented = commented
+
+		try:
+			self.value = eval(value.replace('true', 'True').replace('false', 'False'))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of possibly insecure function - consider using safer ast.literal_eval.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4699682)